### DISCUSSION TITLE
S3 support

### DISCUSSION THREAD

**Discussion by jueshali** [juicedata/juicefs#107] [Q&A] [answered]
Does the current version of JuiceFS support S3? I did not find the S3 API and other related codes in the source code. But there are S3 related tests in the Readme.

**Comment by xiaogaozi** [comment, marked as answer]
S3 related code is in another project called [juicesync](https://github.com/juicedata/juicesync/blob/master/object/s3.go). JuiceFS depends on juicesync.

**Reply by jueshali** [reply]
thx

**Reply by davies** [reply]
btw, we have moved the source code of object support back into juicefs, please check out them here: https://github.com/juicedata/juicefs/tree/main/pkg/object
